### PR TITLE
Fixed tools situations reset fields

### PR DESCRIPTION
**Pull request by nkrmr**
Fixed reset fields when parent changed

### CODE REVIEW

File: packages/code-du-travail-frontend/src/outils/common/situations.utils.js [hunk 1]
```diff
@@ -112,3 +112,23 @@ export function getRef({ ref, refUrl }) {
...
+    // list keys that no longer exist
+    const resetFormProps = Object.keys(initialSituations)
+      .filter(k => !pastQuestions.find(([key]) => k === key))
+      .concat(
+        // list keys that need to be reseted
+        pastQuestions
+          .slice(pastQuestions.findIndex(([k]) => k === key) + 1)
+          .map(([key]) => key)
+      );
+    resetFormProps.forEach(key => form.change(`criteria.${key}`, undefined));
```
Comment (lionelB) [inline packages/code-du-travail-frontend/src/outils/common/situations.utils.js]: `criteria` should be extracted unless we want to forget to update it is forms data change ;)

**Review by lionelB — CHANGES_REQUESTED** [review]
seems alsmost good. just missing a small test

File: packages/code-du-travail-frontend/src/outils/common/situations.utils.js (hunk 1 above)
Comment (nkrmr) [reply packages/code-du-travail-frontend/src/outils/common/situations.utils.js]: Of course

**Review by lionelB — APPROVED** [review]
👌

**Pull request merged**